MODULE.bazel

```diff
@@ -0,0 +1,94 @@
+module(name = "rules_webtesting", version = "0.4.0")
+
+bazel_dep(name = "rules_go", version = "0.51.0-rc1", repo_name = "io_bazel_rules_go")
+bazel_dep(name = "rules_scala", version = "", repo_name = "io_bazel_rules_scala")
+# TODO: Switch to a release version once it's available in BCR.
+# Otherwise, the root module has to declare the same override.
```

Comment:
> Did you have specific reason for this?

This is both velocity and practicality problem:

Many times we see problems during a release and patch other projects to do the release. With this we will be required to wait any fixes to be released before doing our own release. Waiting for a release for things that we don't have control on will be a huge velocity concern (even if we automate/steamline BCR process for our own repos).

Similarly sometimes to avoid larger migration and compatibility issues, we use a commit between official releases that avoid the problem. That will also start being a problem.

Another problem that exaggerates the issue is, we have multiple repos that work on the head. Also we have sub-workspaces to test out the setup as seen from outside of our workspace. And I also expect some users to use our head version (which was common in the past since we don't do frequent releases). When we need to rely on a patch like this for any reason, it will need to be copied over every single workspace where in most cases they don't even know about J2CL dependencies.

> Because dependency resolution won't work 

Would it be possible to declare closest version where the commit is based on? 
So it will be like `N + epsilon` where it will be newer than version `N`, older than `N+1` but not comparable with any other `N + epsilon`s. In a case where it is not comparable the users needs to resolve by override as before. That could significantly reduce the need for user overrides.

Reply:
> Waiting for a release for things that we don't have control on will be a huge velocity concern

One thing we could do is to upstream the fix as a [BCR patch version](https://github.com/bazelbuild/bazel-central-registry/blob/main/docs/README.md#versions-format) if the module maintainer is too slow. Anyway can send a PR and a registry maintainer can approve. But in general, I think at least for popular Bazel rules, they are well supported in BCR.

> we have multiple repos that work on the head. Also we have sub-workspaces to test out the setup as seen from outside of our workspace.

If they are living in the same repo, then they could probably share the same overrides via [include](https://bazel.build/rules/lib/globals/module#include) in MODULE.bazel with some symlink tricks?

> Would it be possible to declare closest version where the commit is based on?

This is big feature request that needs a lot of thoughts, I'm not sure it's a good idea to do that. @Wyverald 

I'm happy to look at some specific cases where you have to use commits instead a released version and see if we can find a better process or should actually change Bzlmod to support it better.